### PR TITLE
add micromasters int and mart tables for exam grades

### PR DESCRIPTION
**Pull request by KatelynGit**
### What are the relevant tickets?
https://github.com/mitodl/hq/issues/3431

### Description (What does it do?)
Creates the marts__micromasters_dedp_exam_grades and int__micromasters__dedp_proctored_exam_grades tables

### How can this be tested?
dbt build --select +marts__micromasters_dedp_exam_grades

### CODE REVIEW

File: src/ol_dbt/models/intermediate/micromasters/_int_micromasters__models.yml [hunk 1]
```diff
@@ -2,6 +2,53 @@
...
+  - name: course_title
+    description: str, title of the course
+  - name: course_number
+    description: str, edX course number
+  - name: user_edxorg_username
+    description: str, username on edX.org
+  - name: user_mitxonline_username
+    description: str, username on MITx Online
+  - name: user_full_name
+    description: str, user full name on edX.org or MITxOnline. Very small number of
```
Comment (rachellougee) [inline src/ol_dbt/models/intermediate/micromasters/_int_micromasters__models.yml:25]: Looking at the query, we are getting `user_full_name` from __micromasters__users, so I am not sure why it says from edX.org or MITxOnline? Can we double check?

Comment (rachellougee) [reply src/ol_dbt/models/intermediate/micromasters/_int_micromasters__models.yml:25]: Same comments as `user_address_country` and `user_email`

File: src/ol_dbt/models/intermediate/micromasters/_int_micromasters__models.yml [hunk 2]
```diff
@@ -2,6 +2,53 @@
...
+      edX.org users have blank full name, their name couldn't be populated from other
+      sources if they don't have their accounts linked on MicroMasters.
+  - name: user_address_country
+    description: str, country code provided by the user from MITx Online or edX.org
+  - name: user_email
+    description: str, current user email on edX.org or MITxOnline. For edx.org, learners
+      can update their emails, so this is not necessary the same email as when learner
+      enrolled for the course
```
Comment (rachellougee) [inline src/ol_dbt/models/intermediate/micromasters/_int_micromasters__models.yml:33]: Beside the description, I think user_email can't be null?

File: src/ol_dbt/models/intermediate/micromasters/_int_micromasters__models.yml [hunk 3]
```diff
@@ -2,6 +2,53 @@
...
+  - name: proctoredexamgrade_percentage_grade
+    description: float, user percentage grade for the course, range between 0 to 1
+  - name: proctoredexamgrade_exam_on
+    description: timestamp, date and time for the exam
+  - name: proctoredexamgrade_created_on
+    description: timestamp, date and time when this grade was initially created
+  - name: proctoredexamgrade_updated_on
+    description: timestamp, date and time when this grade was most recently updated
+  - name: proctoredexamgrade_id
+    description: int, sequential ID representing a DEDP proctor exam grade
```
Comment (rachellougee) [inline src/ol_dbt/models/intermediate/micromasters/_int_micromasters__models.yml:50]: Can we add some test to these fields? null and unique keys? Staging table doesn't indicate that, but I think `grades_proctoredexamgrade` table is static now since it has stopped updating since 2022-09. So we should be able to find out the uniqueness in this table

File: src/ol_dbt/models/marts/micromasters/_marts_micromasters__models.yml [hunk 4]
```diff
@@ -2,6 +2,53 @@
 version: 2
 
 models:
+- name: marts__micromasters_exam_grades
```
Comment (rachellougee) [inline src/ol_dbt/models/marts/micromasters/_marts_micromasters__models.yml:5]: Does it make sense to specify DEDP for the mart name? This table is for old DEDP and has no other MicroMasters exam grades, so i think we should be clear about that in the name?

**Review by rachellougee — COMMENTED** [review]
Mostly feedback on field description and naming for the mart.

File: src/ol_dbt/models/intermediate/micromasters/_int_micromasters__models.yml (hunk 1 above)
Comment (rachellougee) [reply src/ol_dbt/models/intermediate/micromasters/_int_micromasters__models.yml:25]: These profile fields are sourced from `stg__micromasters__app__postgres__profiles_profile`, so these come directly from MicroMasters

File: src/ol_dbt/models/marts/micromasters/_marts_micromasters__models.yml (hunk 4 above)
Comment (rachellougee) [reply src/ol_dbt/models/marts/micromasters/_marts_micromasters__models.yml:5]: @pdpinch not sure what you think of the naming this mart as I believe this is specific for DEDP proctor exam that was last updated in 2022-07

File: src/ol_dbt/models/intermediate/micromasters/_int_micromasters__models.yml (hunk 3 above)
Comment (KatelynGit) [reply src/ol_dbt/models/intermediate/micromasters/_int_micromasters__models.yml:50]: added tests

File: src/ol_dbt/models/intermediate/micromasters/_int_micromasters__models.yml (hunk 1 above)
Comment (KatelynGit) [reply src/ol_dbt/models/intermediate/micromasters/_int_micromasters__models.yml]: I changed these descriptions

File: src/ol_dbt/models/intermediate/micromasters/_int_micromasters__models.yml (hunk 2 above)
Comment (KatelynGit) [reply src/ol_dbt/models/intermediate/micromasters/_int_micromasters__models.yml]: added test

File: src/ol_dbt/models/marts/micromasters/_marts_micromasters__models.yml (hunk 4 above)
Comment (KatelynGit) [reply src/ol_dbt/models/marts/micromasters/_marts_micromasters__models.yml]: changed the table names

File: src/ol_dbt/models/intermediate/micromasters/_int_micromasters__models.yml [hunk 5]
```diff
@@ -2,6 +2,56 @@
 version: 2
 
 models:
+- name: int__micromasters__exam_grades_old_data
```
Comment (pdpinch) [inline src/ol_dbt/models/intermediate/micromasters/_int_micromasters__models.yml:5]:
```suggestion
- name: int__micromasters__dedp_proctored_exam_grades
```

I agree with Rachel that this table name could be clearer. If this is too long, we could remove the word `proctored`

File: src/ol_dbt/models/intermediate/micromasters/_int_micromasters__models.yml [hunk 6]
```diff
@@ -2,6 +2,56 @@
...
+    tests:
+    - not_null
+  - name: micromasters_program_id
+    description: int, id of the program in the micromasters database
+  - name: mitxonline_program_id
+    description: int, id of the program in the mitxonline database
+  - name: course_title
+    description: str, title of the course
+  - name: course_number
+    description: str, edX course number
```
Comment (pdpinch) [inline src/ol_dbt/models/intermediate/micromasters/_int_micromasters__models.yml:19]: It would be useful to have an example in the description here.

File: src/ol_dbt/models/intermediate/micromasters/_int_micromasters__models.yml [hunk 7]
```diff
@@ -2,6 +2,56 @@
...
+  - name: user_edxorg_username
+    description: str, username on edX.org
+  - name: user_mitxonline_username
+    description: str, username on MITx Online
+  - name: user_full_name
+    description: str, user full name on micromasters
+  - name: user_address_country
+    description: str, country code provided by the user from micromasters
```
Comment (pdpinch) [inline src/ol_dbt/models/intermediate/micromasters/_int_micromasters__models.yml:27]: This field seems out of place to me here. Is there a reason why we have it?

File: src/ol_dbt/models/intermediate/micromasters/_int_micromasters__models.yml [hunk 8]
```diff
@@ -2,6 +2,56 @@
 version: 2
 
 models:
+- name: int__micromasters__exam_grades_old_data
+  description: DEDP proctor exam grades stored in MicroMasters database
```
Comment (pdpinch) [inline src/ol_dbt/models/intermediate/micromasters/_int_micromasters__models.yml:6]: We could mention in the description that there haven't been any new proctored exam grades since `insert date`

File: src/ol_dbt/models/marts/micromasters/_marts_micromasters__models.yml [hunk 9]
```diff
@@ -2,6 +2,56 @@
 version: 2
 
 models:
+- name: marts__micromasters_exam_grades_old_data
+  description: DEDP proctor exam grades stored in MicroMasters database
```
Comment (pdpinch) [inline src/ol_dbt/models/marts/micromasters/_marts_micromasters__models.yml:6]: name and description should correspond to the intermediate table name and description.

Comment (pdpinch) [reply src/ol_dbt/models/marts/micromasters/_marts_micromasters__models.yml:6]: That said... I'm not 100% sure we need a mart for this table. If we meet with the DEDP team and learn that they would prefer this data a different way -- like maybe as part of the certificate mart -- can we remove this later?

**Review by pdpinch — CHANGES_REQUESTED** [review]

File: src/ol_dbt/models/marts/micromasters/_marts_micromasters__models.yml (hunk 4 above)
Comment (KatelynGit) [reply src/ol_dbt/models/marts/micromasters/_marts_micromasters__models.yml]: i changed the name

File: src/ol_dbt/models/intermediate/micromasters/_int_micromasters__models.yml [hunk 10]
```diff
@@ -2,6 +2,56 @@
...
+    - not_null
+  - name: micromasters_program_id
+    description: int, id of the program in the micromasters database
+  - name: mitxonline_program_id
+    description: int, id of the program in the mitxonline database
+  - name: course_title
+    description: str, title of the course for example Foundations of Modern Finance
+      I
+  - name: course_number
+    description: str, edX course number for example CTL.SC4x or 15.415.1x
```
Comment (pdpinch) [inline src/ol_dbt/models/intermediate/micromasters/_int_micromasters__models.yml:21]: since this table only applies to DEDP, we might as well use DEDP examples. 

```suggestion
    description: str, edX course number for example 14.73x or JPAL102x
```

**Review by rachellougee — APPROVED** [review]

File: src/ol_dbt/models/marts/micromasters/_marts_micromasters__models.yml [hunk 11]
```diff
@@ -2,6 +2,57 @@
...
+  - name: user_edxorg_username
+    description: str, username on edX.org
+  - name: user_mitxonline_username
+    description: str, username on MITx Online
+  - name: user_full_name
+    description: str, user full name on MicroMasters
+  - name: user_address_country
+    description: str, country code provided by the user from MicroMasters
+  - name: user_email
+    description: str, current user email on MicroMasters
```
Comment (pdpinch) [inline src/ol_dbt/models/marts/micromasters/_marts_micromasters__models.yml:30]: Same comment as above. If this email is coming from micromasters we should make that clear. 

Is it possible to include a mitxonline email as well, for those situations when they are different?

File: src/ol_dbt/models/marts/micromasters/_marts_micromasters__models.yml [hunk 12]
```diff
@@ -2,6 +2,57 @@
...
+    description: float, user score for the course, range between 0 to 100
+  - name: proctoredexamgrade_is_passing
+    description: boolean, indicating whether the user has passed the passing score
+      set for this course
+  - name: proctoredexamgrade_passing_score
+    description: float, passing score for the course, range between 0 to 100
+  - name: proctoredexamgrade_percentage_grade
+    description: float, user percentage grade for the course, range between 0 to 1
+  - name: proctoredexamgrade_exam_on
+    description: timestamp, date and time for the exam
```
Comment (pdpinch) [inline src/ol_dbt/models/marts/micromasters/_marts_micromasters__models.yml:43]: Same comment as above. Is this the timestamp for when the learner took the exam? If so, make it clearer.

**Review by pdpinch — CHANGES_REQUESTED** [review]

File: src/ol_dbt/models/intermediate/micromasters/_int_micromasters__models.yml [hunk 13]
```diff
@@ -2,6 +2,56 @@
...
+  - name: course_number
+    description: str, edX course number for example 14.73x or JPAL102x
+  - name: user_edxorg_username
+    description: str, username on edX.org
+  - name: user_mitxonline_username
+    description: str, username on MITx Online
+  - name: user_full_name
+    description: str, user full name on micromasters
+  - name: user_email
+    description: str, current user email on micromasters
```
[the remark this answers is not in the document]
Comment (KatelynGit) [reply src/ol_dbt/models/intermediate/micromasters/_int_micromasters__models.yml]: I've changed this to user_micromasters_email

File: src/ol_dbt/models/intermediate/micromasters/_int_micromasters__models.yml [hunk 14]
```diff
@@ -2,6 +2,56 @@
...
+    description: float, user score for the course, range between 0 to 100
+  - name: proctoredexamgrade_is_passing
+    description: boolean, indicating whether the user has passed the passing score
+      set for this course
+  - name: proctoredexamgrade_passing_score
+    description: float, passing score for the course, range between 0 to 100
+  - name: proctoredexamgrade_percentage_grade
+    description: float, user percentage grade for the course, range between 0 to 1
+  - name: proctoredexamgrade_exam_on
+    description: timestamp, date and time for the exam
```
[the remark this answers is not in the document]
Comment (KatelynGit) [reply src/ol_dbt/models/intermediate/micromasters/_int_micromasters__models.yml]: I added to the description that it's the date the learner took the exam

File: src/ol_dbt/models/marts/micromasters/_marts_micromasters__models.yml [hunk 15]
```diff
@@ -2,6 +2,57 @@
...
+    tests:
+    - not_null
+  - name: micromasters_program_id
+    description: int, id of the program in the micromasters database
+  - name: mitxonline_program_id
+    description: int, id of the program in the mitxonline database
+  - name: course_title
+    description: str, title of the course
+  - name: course_number
+    description: str, edX course number
```
[the remark this answers is not in the document]
Comment (KatelynGit) [reply src/ol_dbt/models/marts/micromasters/_marts_micromasters__models.yml]: I've changed this to the suggested description

File: src/ol_dbt/models/marts/micromasters/_marts_micromasters__models.yml (hunk 11 above)
Comment (KatelynGit) [reply src/ol_dbt/models/marts/micromasters/_marts_micromasters__models.yml]: I've added the mixonline email and changed the other field to contain the word micromasters

File: src/ol_dbt/models/marts/micromasters/_marts_micromasters__models.yml (hunk 12 above)
Comment (KatelynGit) [reply src/ol_dbt/models/marts/micromasters/_marts_micromasters__models.yml]: I've changed this per the suggestion

**Comment by KatelynGit** [comment]
@pdpinch I think this should be all set for sign off with the changes but let me know if not

**Review by pdpinch — APPROVED** [review]
